### PR TITLE
sc: init at 2024-08-15

### PR DESCRIPTION
**Pull request by claes**
## Description of changes

sc is a curses spreadsheet program. 
It is similar to sc-im but has earlier origin.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sc/sc/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  src = fetchFromGitHub {
+    repo = "sc";
+    owner = "n-t-roff";
+    rev = "e029bc0fb5fa29da1fd23b04fa2a97039a96d2ba";
+    hash = "sha256-JQY+ixHL+TpP4YRpgB9GP4jO5+PBMS/v5Ad3Ux0+yuQ=";
+  };
+
+  buildInputs = [ ncurses ];
+
+  nativeBuildInputs = [ yacc ];
```
Comment (markuskowa) [inline pkgs/by-name/sc/sc/package.nix:22]:
```suggestion
  nativeBuildInputs = [ bison ];
```
It should probably be bison (GNU yacc) instead of yacc.

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sc/sc/package.nix (hunk 1 above)
Comment (claes) [reply pkgs/by-name/sc/sc/package.nix:21]: It didn't work so reverted with new force-push

Comment (markuskowa) [reply pkgs/by-name/sc/sc/package.nix:21]: I can build it with bison (you also need to change it above in the inputs yacc -> bison).

**Comment by markuskowa** [comment]
@GrahamcOfBorg build sc

**Review by markuskowa — APPROVED** [review]
Congrats to your first PR!

**Comment by claes** [comment]
Fantastic, what a successful collaboration!

**Pull request merged**